### PR TITLE
Security enhancement

### PR DESCRIPTION
**Pull request by yogevh**

### CODE REVIEW

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 1]
```diff
@@ -38,12 +39,19 @@ def lambda_handler(event, context):
             requestKeyPairName = event['ResourceProperties']['KeyPairName']
             requestAWSRegionName = event['ResourceProperties']['AWSRegionName']
             requestAWSAccountId = event['ResourceProperties']['AWSAccountId']
+            requestS3BucketName = event['ResourceProperties']['S3BucketName']
+            requestPublicKeyName = event['ResourceProperties']['PVWAVerificationKeyFileName']
```
Comment (avishayil) [inline src/aws_environment_setup/AWSEnvironmentSetup.py]: Change "Public Key" to "Verification Key"

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 2]
```diff
@@ -38,12 +39,19 @@ def lambda_handler(event, context):
...
 
-            isPasswordSaved = save_password_to_param_store(requestPassword)
+            isPasswordSaved = save_password_to_param_store(requestPassword, "AOB_Vault_Pass", "Vault Password")
             if not isPasswordSaved:  # if password failed to be saved
                 return cfnresponse.send(event, context, cfnresponse.FAILED, "Failed to create Vault user's password in Parameter Store",
                                         {}, physicalResourceId)
 
+            isPublicKeySaved = save_public_key_to_param_store(requestS3BucketName, requestPublicKeyName)
```
Comment (avishayil) [inline src/aws_environment_setup/AWSEnvironmentSetup.py]: Change "Public Key" to "Verification Key"

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 3]
```diff
@@ -38,12 +39,19 @@ def lambda_handler(event, context):
...
+            if not isPublicKeySaved:  # if password failed to be saved
```
Comment (avishayil) [inline src/aws_environment_setup/AWSEnvironmentSetup.py]: Change "Public Key" to "Verification Key"

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 4]
```diff
@@ -284,18 +291,29 @@ def create_session_table():
     print("Table 'Sessions' created successfully")
     return True
 
+def save_public_key_to_param_store(S3BucketName, PublicKeyName):
```
Comment (avishayil) [inline src/aws_environment_setup/AWSEnvironmentSetup.py]: Change "Public Key" to "Verification Key"

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 5]
```diff
@@ -284,18 +291,29 @@ def create_session_table():
...
+    try:
+        s3Resource = boto3.resource('s3')
+        s3Resource.Bucket(S3BucketName).download_file(PublicKeyName, '/tmp/server.crt')
```
Comment (avishayil) [inline src/aws_environment_setup/AWSEnvironmentSetup.py]: Change "Public Key" to "Verification Key"

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 6]
```diff
@@ -284,18 +291,29 @@ def create_session_table():
...
+        save_password_to_param_store(open('/tmp/server.crt').read(),"AOB_PVWA_Verification_Key","PVWA Public Key")
```
Comment (avishayil) [inline src/aws_environment_setup/AWSEnvironmentSetup.py]: Change "Public Key" to "Verification Key"

File: src/aws_environment_setup/AWSEnvironmentSetup.py [hunk 7]
```diff
@@ -284,18 +291,29 @@ def create_session_table():
...
+    except Exception as e:
+        print("An error occurred while downloading PublicKey from S3 Bucket - {0}. Exception: {1}".format(S3BucketName, e))
```
Comment (avishayil) [inline src/aws_environment_setup/AWSEnvironmentSetup.py]: Change "Public Key" to "Verification Key"

File: dist/multi-region-cft/CyberArk-AOB-MultiRegion-CF.json [hunk 8]
```diff
@@ -1,1151 +1,1100 @@
...
+            "ElasticityLambda",
+            "Arn"
+          ]
+        },
+        "Principal": "sns.amazonaws.com",
+        "SourceArn": {
+          "Fn::Join": [
+            "",
+            [
+              "arn:aws:sns:us-east-1:",
```
Comment (avishayil) [inline dist/multi-region-cft/CyberArk-AOB-MultiRegion-CF.json]: Why there is a hardcoded region here?

Comment (avishayil) [reply dist/multi-region-cft/CyberArk-AOB-MultiRegion-CF.json]: Edit: I see it in all regions, just explain

Comment (yogevh) [reply dist/multi-region-cft/CyberArk-AOB-MultiRegion-CF.json]: we are creating the infrastructure for the solution to be supported in all regions
the next CF StackSet will create the second neccessary permission for the SNS and Elasticity Lambda

**Pull request merged**